### PR TITLE
Add infix feature extractors samples (#1568)

### PR DESCRIPTION
**Pull request by constantine-zubrylin**
Adds infix feature extractors samples based on fluent tests

______________________________________
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/main/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: apis/fluent/atrium-api-fluent/src/commonTest/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/FeatureExtractorSamples.kt [hunk 1]
```diff
@@ -45,7 +45,7 @@ class FeatureExtractorSamples {
                 .its({ age }) {
                     // introduces an expectation-group block
                     // all expectations are evaluated inside an expectation-group block; for more details:
-                    // https://github.com/robstoll/atrium#define-single-expectations-or-expectation-groups
+                    // https://github.com/robstoll/atrium#define-single-expectations-or-expectation-groups TODO: check link
```
Comment (robstoll) [inline apis/fluent/atrium-api-fluent/src/commonTest/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/FeatureExtractorSamples.kt:48]: :+1: should be https://github.com/robstoll/atrium#define-single-expectations-or-an-expectation-group, I already did a search and replace in main

File: apis/infix/atrium-api-infix/src/commonTest/kotlin/ch/tutteli/atrium/api/infix/en_GB/samples/FeatureExtractorSamples.kt [hunk 2]
```diff
@@ -5,18 +5,538 @@ import ch.tutteli.atrium.api.verbs.expect
...
+        val person = Person(name = "John Smith", age = 25)
+
+        expect(person) {
+            its { age } toBeLessThan 30 toBeGreaterThan 20
+        }
+
+        fails {
+            expect(person) its { age } toBeLessThan 20 toBeGreaterThan 30
+            //                         |               | not evaluated/reported because `toBeLessThan` already fails
+            //                         |               | use an expectation-group block; for more details: https://github.com/robstoll/atrium#define-single-expectations-or-expectation-groups
```
Comment (robstoll) [inline apis/infix/atrium-api-infix/src/commonTest/kotlin/ch/tutteli/atrium/api/infix/en_GB/samples/FeatureExtractorSamples.kt:25]: update link as well

File: apis/infix/atrium-api-infix/src/commonTest/kotlin/ch/tutteli/atrium/api/infix/en_GB/samples/FeatureExtractorSamples.kt [hunk 3]
```diff
@@ -5,18 +5,538 @@ import ch.tutteli.atrium.api.verbs.expect
...
+//            // subject here is back to type Person
+//
+//        fails {
+//            expect(person)
+//                .its { age }
+//                    // introduces an expectation-group block
+//                    // all expectations are evaluated inside an expectation-group block; for more details:
+//                    // https://github.com/robstoll/atrium#define-single-expectations-or-expectation-groups
+//                    .toBeGreaterThan(40) toBeLessThan(50)    // still evaluated, use `.its.` if you want fail fast behaviour // TODO fix comment
+//        }
```
Comment (robstoll) [inline apis/infix/atrium-api-infix/src/commonTest/kotlin/ch/tutteli/atrium/api/infix/en_GB/samples/FeatureExtractorSamples.kt:45]: in most places we place the link right after `fails {` (I fixed some outliers in main). Thus I suggest we use here as well (and in other places in this file):
```suggestion
        fails {
            // all expectations are evaluated inside an expectation-group block; for more details:
            // https://github.com/robstoll/atrium#define-single-expectations-or-an-expectation-group

            expect(person).its({ age }) {
                it toBeGreaterThan(40)
                it toBeLessThan(50)    // still evaluated, use `.its { ... }` if you want fail fast behaviour
            }
        }
```

**Comment by robstoll** [comment]
@constantine-zubrylin just as info, did not have a look again, let me know once it is ready for review

**Comment by constantine-zubrylin** [comment]
> @constantine-zubrylin just as info, did not have a look again, let me know once it is ready for review

Sure. Sorry, I have some problems with free time, but I'll return to the ticket soon.

**Comment by robstoll** [comment]
constantine no longer has time to work on it. as it is a good first issue, the next contributor can as well start from scratch

**Pull request closed without merging**